### DISCUSSION TITLE
How to remove API key rate limit?

### DISCUSSION THREAD

**Discussion by devrsi0n** [plausible/analytics#1573] [Self-Hosted Support] [answered]
Thanks for your hard work, really appreciate it! 👍 

I would like to use [API keys](https://plausible.io/docs/stats-api#custom-props) to access the stats API, but the doc says it has a rate limit, I want to know how to remove it on self-hosted version. Tried to search it in this repo, but didn't find the luck.

Thank you for your time.

![image](https://user-images.githubusercontent.com/7880675/147739588-67144392-9877-4cfd-b8dc-f8fa172e0006.png)

**Comment by ukutaht** [comment, marked as answer]
There's no way to do it through the UI but you can directly change the request limits in the database.

1. Access the postgres database: https://github.com/plausible/analytics/discussions/621#discussioncomment-293169
2. `UPDATE api_keys set hourly_request_limit=99999999`

**Reply by devrsi0n** [reply]
@ukutaht Thank you so much for your help, I also figured it out by searching the keyword `600` in the source code.

Here are my commands to fix it:

Run psql via docker
```bash
docker exec -it plausible_plausible_db_1 psql -U postgres
```

Run psql
```sql
# list databases
\l

# Switching Databases
\c api_keys

# Expanded display
\x

# List the data
SELECT * FROM public.api_keys;

# Update the field with max value of integer
UPDATE public.api_keys SET hourly_request_limit = 2147483647;
```
